packages/nextjs/src/config/loaders/proxyLoader.ts

```diff
@@ -34,11 +36,28 @@ export default async function proxyLoader(this: LoaderThis<LoaderOptions>, userC
     // homepage), sub back in the root route
     .replace(/^$/, '/');
 
+  // For the `excludedServersideEntrypoints` option we need the calculate the relative path to the file in question without file extension.
+  const relativePosixPagePath = path
+    .join('pages', path.relative(pagesDir, this.resourcePath))
+    // Make sure that path is in posix style - this should make it easiser for users to configure and copy & paste from docs
+    .split(path.sep)
+    .join(path.posix.sep)
+    // Pull off the file extension
+    .replace(new RegExp(`\\.(${pageExtensionRegex})`), '');
+
+  const isExcluded = excludedServersideEntrypoints.some(exludeEntry => {
+    if (typeof exludeEntry === 'string') {
+      return relativePosixPagePath === exludeEntry;
+    } else {
+      return relativePosixPagePath.match(exludeEntry);
+    }
+  });
```

Comment:
L: I don't immediately have a great suggestion here, but the fact that we're comparing the excluded paths to the current path rather than vice versa makes my brain hurt a little. (When I put it into words in my head, it feels more intuitive to say "if the current path matches any of the excluded paths" rather than "if any of the excluded paths match the current path.")

Reply:
I honestly wouldn't know how to stop the brain-hurting here :D We only have an array and a string to work with - can we even reverse this logic?